tools/clang/test/HLSLFileCheck/hlsl/types/struct/bitfields_error.hlsl

```diff
@@ -1,6 +1,70 @@
-// RUN: %dxc /T vs_6_0 /E main %s | FileCheck %s
+// RUN: %dxc /T vs_6_2 /E main %s | FileCheck %s
 
-// CHECK: error: bitfields are not supported in HLSL
+// CHECK: error: expected ';' after top level declarator
+typedef int T : 1; // error: typedef member 'E' cannot be a bit-field
+// CHECK: error: expected ';' after top level declarator
+static int sb : 1; // error: static member 'sb' cannot be a bit-field
 
-struct Struct { uint field : 1; };
-void main() {}
\ No newline at end of file
+struct E {
+  // CHECK: error: static member 'bf' cannot be a bit-field
+  static uint bf : 8;
+  // CHECK: error: named bit-field 'pad' has zero width
+  uint pad : 0;
+};
+
+struct S {
+  // CHECK: error: size of bit-field 'b' (33 bits) exceeds size of its type (32 bits)
+  uint b : 33;
+  // CHECK: error: size of bit-field 'c' (39 bits) exceeds size of its type (32 bits)
+  bool c : 39;
+  bool d : 3;
+};
+
+int a[sizeof(S) == 1 ? 1 : -1];
+
+enum E {};
+
+struct Z {};
+typedef int Integer;
+
+struct X {
+  // clang: error: anonymous bit-field
```

Comment:
Various errors are formatted "clang:" rather than "CHECK:", meaning these will not be checked.  Is this intentional?  Why if so?

Reply:
Yes.  I left those to point out where clang expects an error for C++ that does not show up in the HLSL test.